test/e2e/framework/pod/wait.go

```diff
@@ -275,8 +275,8 @@ func WaitForPodCondition(c clientset.Interface, ns, podName, conditionDesc strin
 	if err == nil {
 		return nil
 	}
-	if IsTimeout(err) {
-		e2elog.Logf("Timed out while waiting for pod %s to be %s. Last observed as: %s",
+	if IsTimeout(err) && lastPod != nil {
+		return TimeoutError("timed out while waiting for pod %s to be %s: last observed as:\n%s",
 			podIdentifier(ns, podName), conditionDesc, format.Object(lastPod, 1))
```

Comment:
(no action required, maybe a follow-up) I've already found these pod logs useful for debugging tests, but they are quite large and noisy. I'm thinking about using a yaml serializer instead, so that omitEmpty fields aren't (which would significantly cut down on the output size).

Reply:
I think that would be a useful enhancement for `gomega/format.Object`: skip fields that are marked as optional for JSON or YAML.

Otherwise we'll be stuck between a rock and a hard place when choosing a formatter: YAML serializers typically ignore unexported fields, which is bad for error types. output.Object prints those, but doesn't skip.

/cc @onsi